torch/distributed/device_mesh.py

```diff
@@ -152,12 +156,13 @@ def create_flatten_mesh(self, device_mesh: "DeviceMesh") -> "DeviceMesh":
                 flattened_mesh = DeviceMesh(
                     root_mesh.device_type,
                     mesh_nd,
-                    mesh_dim_names=(flatten_mesh_dim_names,),
+                    mesh_dim_names=(flatten_mesh_dim_name,),
                 )
                 if cur_rank in mesh_nd:
                     res_flattened_mesh = flattened_mesh
             self.child_to_root_mapping[res_flattened_mesh] = root_mesh  # type: ignore[possibly-undefined]
-            self.root_to_flatten_mapping.setdefault(root_mesh, {})[flatten_mesh_dim_names] = res_flattened_mesh  # type: ignore[possibly-undefined]
+            self.root_to_flatten_mapping.setdefault(root_mesh, {})[flatten_mesh_dim_name] = res_flattened_mesh  # type: ignore[possibly-undefined]
+            self.flatten_name_to_root_dims.setdefault(root_mesh, {})[flatten_mesh_dim_name] = tuple(flatten_dims_in_root)  # type: ignore[possibly-undefined]
```

Comment:
It seems that we do not check the existence of `flatten_mesh_dim_name`. If it exist, shouldn't the `create_flatten_mesh()` just returns the cached one?

Reply:
> It seems that we do not check the existence of `flatten_mesh_dim_name`. If it exist, shouldn't the `create_flatten_mesh()` just returns the cached one?

Ye! Check for existence already exists in line 136 - 141.